### ISSUE TITLE
Super-trait operator doesn't work within trait

### ISSUE THREAD

**Issue by b52** [rust-lang/rust#18088]
Consider the following simple trait example:
```rust
pub trait Indexable<T>: Index<uint, T> {
    fn index2(&self, i: uint) -> &T {
        // Doesn't work:
        // self[i]
        
        // Does work
        self.index(&i)
    }
}
```

As you can see by the comments, using the `[]`-operator on `self` doesn't work directly. Instead, one has to call the `index` function manually.
It seems like something is broken with operators and super-traits.

```
$ rustc --version
rustc 0.12.0-nightly (b5ba2f551 2014-10-06 20:27:14 +0000)
```

**Comment by huonw** [comment]
```rust
pub trait Indexable<T>: std::ops::Index<usize, Output = T> {
    fn index2(&self, i: usize) -> &T {
        &self[i]
    }
}
fn main() {}
```
works fine now, adding needstest.

**Comment by sanxiyn** [comment]
Fixed by #36783.
